src/hippopt/robot_planning/expressions/complementarity.py

```diff
@@ -0,0 +1,155 @@
+import casadi as cs
+
+from hippopt.robot_planning.utilities.terrain_descriptor import TerrainDescriptor
+
+
+def dcc_planar_complementarity(
+    terrain: TerrainDescriptor,
+    point_position_name: str = None,
+    height_multiplier_name: str = "kt",
+    point_position_control_name: str = "u_p",
+    options: dict = None,
+    **_
+) -> cs.Function:
+    options = {} if options is None else options
+    point_position_name = (
+        terrain.get_point_position_name()
+        if point_position_name is None
+        else point_position_name
+    )
+    point_position = cs.MX.sym(point_position_name, 3)
+    height_multiplier = cs.MX.sym(height_multiplier_name, 1)
+    point_control = cs.MX.sym(point_position_control_name, 3)
+
+    terrain_orientation_fun = terrain.orientation_function()
+    height_function = terrain.height_function()
+
+    height = height_function(point_position)
+    terrain_orientation = terrain_orientation_fun(point_position)
+
+    planar_multiplier = cs.tanh(height_multiplier * height)
```

Comment:
I gave this a bit of thought. To be honest I am not convinced. According to the logic you are proposing, it seems like when the height is equal to ``saturate_meters``, the resulting value of ``tanh`` should be 0.95. The problem here is that
tanh(a * b) is not equal to tanh(a) * tanh(b) (in fact tanh(1) is close to 0.7). Hence, in the end, the input number does not have a real meaning either.

I would stick to classical multiplier, since it is also coherent with what I wrote in the paper, and eventually let this conversion to the user.

Reply:
Mmh no wait, if you calculate $k_t$ in this way, if $x=0.025$, we have that $\tanh(k_t x) = 0.95$. This is what I meant :)

In any case, no problem to keep things as they are. I like the interpretability given by my backward calculation, but it's just a way to give more intuition in the tuning phase.